### PR TITLE
favor ssh-rsa keyscans

### PR DESCRIPTION
**Pull request by alexcb**
earthly comes pre-configured with ssh-rsa keyscans for github, gitlab,
bitbucket.

GitHub must have recently added a ecdsa-sha2-nistp256 keyscan, which
the crypto package favoured when passing it to our custom HostKeyCallback
function. This PR instructs the callback to favour passing the ssh-rsa
key when it exists.

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

File: buildcontext/gitlookup.go [hunk 1]
```diff
@@ -230,11 +231,24 @@ func (gl *GitLookup) detectProtocol(host string) (protocol gitProtocol, err erro
 		Auth: []ssh.AuthMethod{
 			ssh.PublicKeysCallback(agent.NewClient(sshAgent).Signers),
 		},
+		HostKeyAlgorithms: []string{
```
Comment (andrewsykim) [inline buildcontext/gitlookup.go:234]: So to clarify, this list is the same as supportedHostKeyAlgos but with `ssh.KeyAlgoRSA` first for compatibility?

Comment (alexcb) [reply buildcontext/gitlookup.go:234]: yeah, I took the entire list from `list from golang.org/x/crypto/ssh/common.go supportedHostKeyAlgos` and moved `ssh.KeyAlgoRSA` to the top.

Comment (andrewsykim) [reply buildcontext/gitlookup.go:234]: The list seems different from the one referenced here? https://cs.opensource.google/go/x/crypto/+/ae814b36:ssh/common.go;l=69-81;bpv=0;bpt=0. Could it have changed in a newer version or something?

Comment (alexcb) [reply buildcontext/gitlookup.go:234]: ah ha! yeah mine came from `mod/golang.org/x/crypto@v0.0.0-20200820211705-5c72a883971a/ssh/common.go`.

Ultimately I'm not sure it matters that much in the context of this PR, as we just need to have `ssh.KeyAlgoRSA` at the top.

When we loop back and allow support for multiple pre-configured key scans, I want to remove this list.

**Review by vladaionescu — APPROVED** [review]

File: buildcontext/gitlookup.go (hunk 1 above)
Comment (andrewsykim) [reply buildcontext/gitlookup.go:234]: sgtm, thanks!

**Pull request merged**